cirq/contrib/routing/device.py

```diff
@@ -47,3 +47,30 @@ def gridqubits_to_graph_device(qubits: Iterable[cirq.GridQubit]):
 
 def _manhattan_distance(qubit1: cirq.GridQubit, qubit2: cirq.GridQubit) -> int:
     return abs(qubit1.row - qubit2.row) + abs(qubit1.col - qubit2.col)
+
+
+def nx_qubit_layout(graph: nx.Graph) \
+        -> Dict[Union[cirq.GridQubit, cirq.LineQubit], Tuple[int, int]]:
+    """Return a layout for a graph for nodes which are qubits.
+
+    This can be used in place of nx.spring_layout or other networkx layouts.
+    GridQubits are positioned according to their row/col. LineQubits are
+    positioned in a line.
+
+    >>> import cirq.contrib.routing as ccr
+    >>> import networkx as nx
+    >>> g = ccr.xmon_device_to_graph(cirq.google.Foxtail)
+    >>> pos = ccr.nx_qubit_layout(g)
+    >>> nx.draw_networkx(g, pos=pos)
```

Comment:
This code will be executed by doctest

Reply:
is that a problem?

Reply:
It depends if draw_networkx makes a file in some random folder or not

Reply:
It doesn't [look like](https://networkx.github.io/documentation/stable/_modules/networkx/drawing/nx_pylab.html#draw_networkx) `draw_networkx` writes to any files.